src/typechecker/FStar.TypeChecker.Tc.fs

```diff
@@ -2156,10 +2156,9 @@ and finish_partial_modul (loading_from_cache:bool) (iface_exists:bool) (en:env)
 
     //pop BUT ignore the old env
     pop_context env ("Ending modul " ^ modul.name.str) |> ignore;
-    env.solver.encode_modul env modul;
-    env.solver.refresh();
+
     //interactive mode manages it itself
-    let _ = if not (Options.interactive ()) then Options.restore_cmd_line_options true |> ignore else () in
+    //let _ = if not (Options.interactive ()) then Options.restore_cmd_line_options true |> ignore else () in
```

Comment:
This change looks potentially unrelated ... what's up here?

Reply:
To keep the behavior same, this is now moved to `Universal.fs` after we have encoded the smt module to Z3.